src/webgpu/api/operation/resource_init/check_texture/by_sampling.ts

```diff
@@ -0,0 +1,149 @@
+import { assert } from '../../../../../common/framework/util/util.js';
+import {
+  EncodableTextureFormat,
+  kEncodableTextureFormatInfo,
+} from '../../../../capability_info.js';
+import { getTexelDataRepresentation } from '../../../../util/texture/texelData.js';
+import { CheckContents } from '../texture_zero_init.spec.js';
+
+export const checkContentsBySampling: CheckContents = (
+  t,
+  params,
+  texture,
+  state,
+  subresourceRange
+) => {
+  assert(params.dimension === '2d');
+  assert(params.format in kEncodableTextureFormatInfo);
+  const format = params.format as EncodableTextureFormat;
+  const rep = getTexelDataRepresentation(format);
+
+  for (const { level, slices } of subresourceRange.mipLevels()) {
+    const width = t.textureWidth >> level;
+    const height = t.textureHeight >> level;
+
+    let ReadbackTypedArray:
+      | Float32ArrayConstructor
+      | Int32ArrayConstructor
+      | Uint32ArrayConstructor = Float32Array;
+
+    // TODO: look up component type in texture format tables
+    let shaderType = 'f32';
+    if (params.format.indexOf('sint') !== -1) {
+      shaderType = 'i32';
+      ReadbackTypedArray = Int32Array;
+    } else if (params.format.indexOf('uint') !== -1) {
+      shaderType = 'u32';
+      ReadbackTypedArray = Uint32Array;
+    }
+
+    const componentOrder = rep.componentOrder;
+    const componentCount = componentOrder.length;
+    const indexExpression =
```

Comment:
Can we have a small example of the string that this produces? Is it a swizzle expression? (the variable could be renamed extraSwizzle?). Ah never mind we need the index expression for `vec2<f32>` but not `f32`. Can we have the comment that shows the code still?